### PR TITLE
all: get rid of query pushdown to simplify query path

### PR DESCRIPTION
**Pull request by MichaHoffmann**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

* removed query pushdown code path

## Verification

* previous tests are still passing ( except query pushdown tests )

### CODE REVIEW

File: cmd/thanos/query.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func registerQuery(app *extkingpin.App) {
 
 	activeQueryDir := cmd.Flag("query.active-query-path", "Directory to log currently active queries in the queries.active file.").Default("").String()
 
-	featureList := cmd.Flag("enable-feature", "Comma separated experimental feature names to enable.The current list of features is "+queryPushdown+".").Default("").Strings()
+	featureList := cmd.Flag("enable-feature", "Comma separated experimental feature names to enable.The current list of features is empty.").Default("").Strings()
```
Comment (yeya24) [inline cmd/thanos/query.go:200]: Can we mark this as hidden

Comment (MichaHoffmann) [reply cmd/thanos/query.go]: marked as hidden in latest commit!

**Review by yeya24 — APPROVED** [review]
Thanks

**Review by yeya24 — APPROVED** [review]
Thanks

**Review by fpetkovski — DISMISSED** [review]
Thank you. @GiedriusS are you okay with this change?

File: CHANGELOG.md [hunk 2]
```diff
@@ -47,6 +47,8 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 
 ### Removed
 
+- [#7014](https://github.com/thanos-io/thanos/pull/7014) *: *breaking :warning:* Removed experimental query pushdown feature to simplify query path. This feature has had high complexity for too little benefits.
```
Comment (GiedriusS) [inline CHANGELOG.md:50]: Should we mention that we are working on this in the promql-engine engine? I mean the distributed mode?

File: pkg/api/query/querypb/query.proto [hunk 3]
```diff
@@ -41,6 +41,7 @@ message QueryRequest {
 
   bool enableDedup = 7;
   bool enablePartialResponse = 8;
+  // Deprecated
   bool enableQueryPushdown = 9;
```
Comment (GiedriusS) [inline pkg/api/query/querypb/query.proto:45]: I think the convention is to mark a field as reserved instead of a comment. So, it should be:
`reserved 9;`

**Review by GiedriusS — COMMENTED** [review]
Added some comments. Happy to see this removed :bow:

File: CHANGELOG.md (hunk 2 above)
Comment (MichaHoffmann) [reply CHANGELOG.md:50]: Ill add that the responsibility for this is in the new thanos engine in distributed mode, but i dont feel great about calling it a hack in the changelog!

File: pkg/api/query/querypb/query.proto (hunk 3 above)
Comment (MichaHoffmann) [reply pkg/api/query/querypb/query.proto:45]: TIL about `reserved`! Thanks

**Review by yeya24 — APPROVED** [review]

**Review by GiedriusS — APPROVED** [review]
:medal_military:

**Pull request merged**